crates/storage/provider/src/test_utils/blocks.rs

```diff
@@ -86,57 +97,56 @@ fn block1() -> (SealedBlockWithSenders, PostState) {
     let mut post_state = PostState::default();
     // Transaction changes
     post_state.create_account(
+        number,
+        H160([0x61; 20]),
+        Account { nonce: 1, balance: U256::from(10), bytecode_hash: None },
+    );
+    post_state.create_account(
+        number,
         H160([0x60; 20]),
         Account { nonce: 1, balance: U256::from(10), bytecode_hash: None },
     );
     post_state.change_storage(
+        number,
         H160([0x60; 20]),
         BTreeMap::from([(U256::from(5), (U256::ZERO, U256::from(10)))]),
     );
-    post_state.finish_transition();
-    // Block changes
-    post_state.create_account(
-        H160([0x61; 20]),
-        Account { nonce: 1, balance: U256::from(10), bytecode_hash: None },
-    );
-    post_state.finish_transition();
 
     (SealedBlockWithSenders { block, senders: vec![H160([0x30; 20])] }, post_state)
 }
 
 /// Block two that points to block 1
-fn block2() -> (SealedBlockWithSenders, PostState) {
+fn block2(number: BlockNumber, parent_hash: H256) -> (SealedBlockWithSenders, PostState) {
     let mut block_rlp = hex!("f9025ff901f7a0c86e8cc0310ae7c531c758678ddbfd16fc51c8cef8cec650b032de9869e8b94fa01dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347942adc25665018aa1fe0e6bc666dac8fc2697ff9baa050554882fbbda2c2fd93fdc466db9946ea262a67f7a76cc169e714f105ab583da00967f09ef1dfed20c0eacfaa94d5cd4002eda3242ac47eae68972d07b106d192a0e3c8b47fbfc94667ef4cceb17e5cc21e3b1eebd442cebb27f07562b33836290db90100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008302000001830f42408238108203e800a00000000000000000000000000000000000000000000000000000000000000000880000000000000000f862f860800a83061a8094095e7baea6a6c7c4c2dfeb977efac326af552d8780801ba072ed817487b84ba367d15d2f039b5fc5f087d0a8882fbdf73e8cb49357e1ce30a0403d800545b8fc544f92ce8124e2255f8c3c6af93f28243a120585d4c4c6a2a3c0").as_slice();
     let mut block = SealedBlock::decode(&mut block_rlp).unwrap();
     block.withdrawals = Some(vec![Withdrawal::default()]);
     let mut header = block.header.clone().unseal();
-    header.number = 2;
+    header.number = number;
     header.state_root =
-        H256(hex!("90101a13dd059fa5cca99ed93d1dc23657f63626c5b8f993a2ccbdf7446b64f8"));
+        H256(hex!("474d43027ff7aaa2659c582c91c33536c80c5cbc56a114d1f53835f8cae52125"));
```

Comment:
Why did we need to change the state root here?

Reply:
I wanted to have two changes and introduced the second one that wasn't there, will revert back to have the same state root.